### PR TITLE
Add types for compute pipeline

### PR DESCRIPTION
**Pull request by burtonageo**

### CODE REVIEW

File: src/pipeline/compute.rs [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+        }
+    }
+
+    pub fn set_thread_group_size_is_multiple_of_thread_execution_width(&self, size_is_multiple_of_width: bool) {
+        unsafe {
+            msg_send![self, setThreadGroupSizeIsMultipleOfThreadExecutionWidth:size_is_multiple_of_width]
+        }
+    }
+
+    pub fn stage_input_descriptor(&self) -> Option<&StageInputOutputDescriptor> {
```
Comment (JohnColanduoni) [inline src/pipeline/compute.rs]: Should be `&StageInputOutputDescriptorRef`

File: src/pipeline/compute.rs [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+        unsafe {
+            msg_send![self, stageInputDescriptor]
+        }
+    }
+
+    pub fn set_stage_input_descriptor(&self, descriptor: Option<&StageInputOutputDescriptor>) {
```
Comment (JohnColanduoni) [inline src/pipeline/compute.rs]: Should be `&StageInputOutputDescriptorRef`

File: src/pipeline/compute.rs [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+pub enum MTLStageInputOutputDescriptor {}
+
+foreign_obj_type! {
+    type CType = MTLStageInputOutputDescriptor;
+    pub struct StageInputOutputDescriptor;
+    pub struct StageInputOutputDescriptorRef;
+}
+
+impl StageInputOutputDescriptor {
+    pub fn new() -> Self {
```
Comment (JohnColanduoni) [inline src/pipeline/compute.rs]: This should return `&StageInputOutputDescriptorRef` (functions with names that don't start with `new`, `create`, `make`, or `alloc` usually return unretained pointers).

File: src/pipeline/compute.rs [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+
+pub enum MTLComputePipelineDescriptor {}
+
+foreign_obj_type! {
+    type CType = MTLComputePipelineDescriptor;
+    pub struct ComputePipelineDescriptor;
+    pub struct ComputePipelineDescriptorRef;
+}
+
+impl ComputePipelineDescriptorRef {
```
Comment (JohnColanduoni) [inline src/pipeline/compute.rs]: I think you're missing a `new` for `ComputePipelineDescriptor`.

**Comment by kvark** [comment]
@burtonageo ping. Would you want to finish these changes?

**Comment by grovesNL** [comment]
We can close this now due to #22

**Pull request closed without merging**